src/app/public/modules/text-editor/text-editor.component.spec.ts

```diff
@@ -0,0 +1,1177 @@
+import {
+  CommonModule
+} from '@angular/common';
+
+import {
+  ComponentFixture,
+  TestBed,
+  fakeAsync,
+  tick
+} from '@angular/core/testing';
+
+import {
+  FormsModule
+} from '@angular/forms';
+
+import {
+  RouterTestingModule
+} from '@angular/router/testing';
+
+import {
+  SkyModalService
+} from '@skyux/modals';
+
+import {
+  SkyDropdownModule
+} from '@skyux/popovers';
+
+import {
+  SkyThemeService
+} from '@skyux/theme';
+
+import {
+  expect,
+  expectAsync,
+  SkyAppTestUtility
+} from '@skyux-sdk/testing';
+
+import {
+  STYLE_STATE_DEFAULTS
+} from './defaults/style-state-defaults';
+
+import {
+  TextEditorFixtureComponent
+} from './fixtures/text-editor.component.fixture';
+
+import {
+  SkyTextEditorAdapterService
+} from './services/text-editor-adapter.service';
+
+import {
+  SkyTextEditorService
+} from './services/text-editor.service';
+
+import {
+  SkyTextEditorSelectionService
+} from './services/text-editor-selection.service';
+
+import {
+  SkyTextEditorMenubarAction
+} from './types/menubar-action';
+
+import {
+  SkyTextEditorStyleState
+} from './types/style-state';
+
+import {
+  SkyTextEditorToolbarActions
+} from './types/toolbar-action';
+
+import {
+  SkyTextEditorModule
+} from './text-editor.module';
+
+describe('Text editor', () => {
+
+  let fixture: ComponentFixture<TextEditorFixtureComponent>;
+
+  //#region helpers
+  function checkboxExecCommandTest(checkboxInputElement: HTMLElement, expectedCommand: string) {
```

Comment:
In all technicality - these should all have return types.

Reply:
Done.